### PR TITLE
Fix issue setting up JSON formatted logging

### PR DESCRIPTION
**Pull request by Patrik-Stas**
- Misplaces imports (which contained logger imports) caused logger to be loaded before logging configuration has been loaded from env variables, consequently disregarding any logging configuration.

- Also additionally refactored boilerplate and config improved config loading and validation - no need to specify which env variables to use to build app config, instead get loaded based on declared of configuration validation.

- Added missing license headers

Signed-off-by: Patrik Stas <patrik.stas@absa.africa>

### CODE REVIEW

File: vcxagency-node/src/logging/logger-builder.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ const winston = require('winston')
 const path = require('path')
 const { jsonFormatter, tryAddRequestId, tryAddEcsTaskMetadata } = require('./logger-common')
 
+console.log(`DISABLE_COLOR_LOGS= ${global.DISABLE_COLOR_LOGS}`)
```
Comment (mirgee) [inline vcxagency-node/src/logging/logger-builder.js:23]: I am not sure about this log, if we are to leave it let's definitely make it `debug` and `DISABLE_COLOR_LOGS=${global.DISABLE_COLOR_LOGS}`

File: vcxagency-node/src/index.js [hunk 2]
```diff
@@ -15,91 +15,75 @@
...
-    logger.info('Attempting to fetch and store ECS task metadata')
-    global.ecsTaskMetadata = await fetchEcsTaskMetadata(appConfig.ECS_CONTAINER_METADATA_URI_V4)
-
-    logger.debug('Going to try reloading trusted certificates.')
-    reloadTrustedCerts(logger)
+async function setGlobalVariables (appConfig) {
+  setLoggingGlobalSettings(appConfig)
+  global.WEBHOOK_RESPONSE_TIMEOUT_MS = appConfig.WEBHOOK_RESPONSE_TIMEOUT_MS
+  const { fetchEcsTaskMetadata } = require('./scripts/fetch-ecs-task-metadata')
+  global.ecsTaskMetadata = await fetchEcsTaskMetadata(appConfig.ECS_CONTAINER_METADATA_URI_V4)
```
Comment (mirgee) [inline vcxagency-node/src/index.js:40]: Similar comment I had in your other PR

File: vcxagency-node/src/index.js [hunk 3]
```diff
@@ -15,91 +15,75 @@
...
+}
 
-    logger.info(`Loaded effective config: ${stringifyAndHideSensitive(appConfig)}`)
+async function run () {
+  const envConfig = await loadConfiguration()
+  const appConfig = await validateAppConfig(envConfig)
+  await setGlobalVariables(appConfig)
```
Comment (mirgee) [inline vcxagency-node/src/index.js:46]: Again, my eyes would like a newline here

File: vcxagency-node/src/index.js [hunk 4]
```diff
@@ -15,91 +15,75 @@
...
-  const appConfig = await loadConfiguration()
-  return startApp(appConfig)
+  reloadTrustedCerts()
+  await startServer(appConfig)
+  return false
 }
 
 run()
+  .then((shouldExit) => {
+    if (shouldExit) {
```
Comment (mirgee) [inline vcxagency-node/src/index.js:83]: Does `run()` return `true` anywhere?

**Review by mirgee — CHANGES_REQUESTED** [review]

File: vcxagency-node/src/logging/logger-builder.js (hunk 1 above)
Comment (Patrik-Stas) [reply vcxagency-node/src/logging/logger-builder.js:23]: yup this was forgotten, deleted now

File: vcxagency-node/src/index.js (hunk 4 above)
Comment (Patrik-Stas) [reply vcxagency-node/src/index.js:83]: No so far for agency.

File: vcxagency-node/src/index.js (hunk 2 above)
Comment (Patrik-Stas) [reply vcxagency-node/src/index.js:40]: dtto reply ^_^

File: vcxagency-node/src/index.js (hunk 4 above)
Comment (mirgee) [reply vcxagency-node/src/index.js:84]: What is the point in unreachable code? :P

**Pull request merged**